tests/test_output.py

```diff
@@ -35,29 +36,24 @@ def capture_print(function_call, *args, **kwargs):
     with redirect_stdout(f):
         function_call(*args, **kwargs)
     out = f.getvalue()
+
+    out_f = 'out_config.ini'
+    if isfile(out_f):
+        os.remove(out_f)
     return out
 
 
-class TestOutput(unittest.TestCase):
+class TestOutput:
 
-    @classmethod
-    def setUpClass(self):
-        base = os.path.dirname(__file__)
-        self.ucfg = get_user_config(os.path.join(base, "test_configs/full_config.ini"),
-                                    modules="inicheck")
+    @pytest.fixture()
+    def ucfg(self, full_config_ini):
+        return get_user_config(full_config_ini, modules="inicheck")
 
-    @classmethod
-    def tearDownClass(self):
-        """
-        Delete any files
+    def test_generate_config_header(self, ucfg):
         """
-        os.remove('out_config.ini')
-
-    def test_generate_config(self):
-        """
-        Tests if we generate a config to a file
-        """
-        generate_config(self.ucfg, 'out_config.ini', cli=False)
+            Tests if we generate a config and dump it to a file with a header
+            """
+        generate_config(ucfg, 'out_config.ini', cli=False)
```

Comment:
This is personal design preference:
Once I have a file name and it has a repeated use like it has in this file, I use a constant at the top of the file. To me, this reads cleaner and also creates a central place to update (if there is a need)

Reply:
Yea agreed. I just made it a fixture to handle the file clean up too.